os/arch/arm/src/armv7-m/up_schedyield.c

```diff
@@ -73,6 +73,11 @@ void up_schedyield(void)
 		return;
 	}
 
+#ifdef CONFIG_TASK_SCHED_HISTORY
+	/* Save the task name which is yielding cpu resource */
+
+	save_task_scheduling_status(rtcb);
```

Comment:
This needs to be moved into the `if (rtcb->sched_priority <= ntcb->sched_priority)` condition below.